sdk/aws/ec2.py

```diff
@@ -87,47 +98,158 @@ def list_instances(self, params_dict=None):
         # return a dictionary that contains the instances_info array and the count of server instances
         return {"count": len(instances_info), "instances": instances_info}
 
-    def create_instance(
-        self, image_id, instance_type, key_name, security_group_id, subnet_id
-    ):
+    def _get_custom_vpc_id(self, vpc_name="openshift-sustaining-vpc"):
+        """
+        Get the custom VPC ID based on the VPC Name or Tag.
+        """
+        ec2_client = self.session.client("ec2")
+
+        try:
+            # Describe VPCs and filter by Name tag (or any other criteria)
+            response = ec2_client.describe_vpcs(
+                Filters=[{"Name": "tag:Name", "Values": [vpc_name]}]
+            )
+            vpcs = response.get("Vpcs", [])
+            if vpcs:
+                return vpcs[0]["VpcId"]
+            else:
+                raise Exception(f"No VPC found with name/tag '{vpc_name}'.")
+        except Exception as e:
+            logger.error(f"Error fetching custom VPC ID: {e}")
+            raise e
+
+    def _get_subnet_ids(self, vpc_id):
+        """
+        Get the subnet IDs associated with the given VPC.
+        """
+        ec2_client = self.session.client("ec2")
+
+        try:
+            # Describe subnets in the VPC
+            response = ec2_client.describe_subnets(
+                Filters=[{"Name": "vpc-id", "Values": [vpc_id]}]
+            )
+
+            # Extract subnet IDs from the response
+            subnet_ids = [subnet["SubnetId"] for subnet in response["Subnets"]]
+            logger.info(f"Found subnets: {subnet_ids}")  # Log the subnets found
+            return subnet_ids
+
+        except Exception as e:
+            logger.error(f"Error fetching subnet IDs: {e}")
+            raise e
+
+    def _get_security_group_id(self, sec_group_name):
+        """
+        Get the security group ID for the VPC.
+        """
+        ec2_client = self.session.client("ec2")
+
+        try:
+            # Describe security groups in the VPC
+            response = ec2_client.describe_security_groups(
+                Filters=[{"Name": "tag:Name", "Values": [f"{sec_group_name}"]}]
+            )
+            sg_id = response["SecurityGroups"][0]["GroupId"]
+            logger.info(
+                f"Found Security Group: {sg_id}"
+            )  # Log the security group found
+            return sg_id
```

Comment:
```suggestion
            response = ec2_client.describe_security_groups(
                Filters=[{"Name": "tag:Name", "Values": [f"{sec_group_name}"]}]
            )
            security_groups = response["SecurityGroups"]
            if not security_groups:
                log.warn("No security groups found")
                return None
            sg_id = security_groups[0]["GroupId"]
            logger.info(
                f"Found Security Group: {sg_id}"
            )  # Log the security group found
            return sg_id
```
recommended to check if the security groups is not empty